oap-server/server-fetcher-plugin/kafka-fetcher-plugin/src/test/java/org/apache/skywalking/oap/server/analyzer/agent/kafka/provider/KafkaFetcherModuleStartIT.java

```diff
@@ -0,0 +1,111 @@
+import org.apache.skywalking.oap.log.analyzer.provider.log.LogAnalyzerServiceImpl;
+import org.apache.skywalking.oap.server.analyzer.agent.kafka.module.KafkaFetcherConfig;
+import org.apache.skywalking.oap.server.analyzer.module.AnalyzerModule;
+import org.apache.skywalking.oap.server.analyzer.provider.AnalyzerModuleConfig;
+import org.apache.skywalking.oap.server.analyzer.provider.AnalyzerModuleProvider;
+import org.apache.skywalking.oap.server.analyzer.provider.trace.parser.ISegmentParserService;
+import org.apache.skywalking.oap.server.analyzer.provider.trace.parser.SegmentParserServiceImpl;
+import org.apache.skywalking.oap.server.core.CoreModule;
+import org.apache.skywalking.oap.server.core.CoreModuleProvider;
+import org.apache.skywalking.oap.server.core.source.SourceReceiver;
+import org.apache.skywalking.oap.server.core.source.SourceReceiverImpl;
+
+import org.apache.skywalking.oap.server.library.module.ModuleManager;
+import org.apache.skywalking.oap.server.library.module.ModuleStartException;
+import org.apache.skywalking.oap.server.telemetry.TelemetryModule;
+import org.apache.skywalking.oap.server.telemetry.api.MetricsCreator;
+import org.apache.skywalking.oap.server.telemetry.none.MetricsCreatorNoop;
+import org.apache.skywalking.oap.server.telemetry.none.NoneTelemetryProvider;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.powermock.reflect.Whitebox;
+import org.testcontainers.containers.KafkaContainer;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+import org.testcontainers.utility.DockerImageName;
+
+@Testcontainers
+@ExtendWith(MockitoExtension.class)
+public class KafkaFetcherModuleStartIT {
+
+    @Mock
+    private NoneTelemetryProvider telemetryProvider;
+    @Mock
+    private CoreModuleProvider coreModuleProvider;
+
+    @Mock
+    private LogAnalyzerModuleProvider logAnalyzerModuleProvider;
+    @Mock
+    private AnalyzerModuleProvider analyzerModuleProvider;
+    @Mock
+    private ModuleManager moduleManager;
+    @Container
+    KafkaContainer container = new KafkaContainer(DockerImageName.parse("confluentinc/cp-kafka:7.1.0-1-ubi8"));
+
+    @BeforeEach
+    public void init() {
+        Mockito.when(telemetryProvider.getService(MetricsCreator.class))
+                .thenReturn(new MetricsCreatorNoop());
+        TelemetryModule telemetryModule = Mockito.spy(TelemetryModule.class);
+        Whitebox.setInternalState(telemetryModule, "loadedProvider", telemetryProvider);
+        Mockito.when(moduleManager.find(TelemetryModule.NAME)).thenReturn(telemetryModule);
+
+        CoreModule coreModule = Mockito.spy(CoreModule.class);
+        Mockito.when(coreModuleProvider.getService(SourceReceiver.class)).thenReturn(new SourceReceiverImpl());
```

Comment:
I believe a lot of mock codes are repeated in many UTs, right? 

Could you move these mock codes into `server-testing`?

Reply:
Meanwhile, due to all these tests are for Graal mostly, as you would not put the verification codes in these UTs, right?
You should reference UT-IT, and separate these UTs to run in the Graal package(such as GRAAL-PACKAGE-IT) only.
For now, you seem to make every UT-IT developer suffer extra time for nothing.